zeeguu/core/feed_handler/newspaperfeed.py

```diff
@@ -0,0 +1,54 @@
+import newspaper
+from .feed_handler import FeedHandler
+
+from zeeguu.logging import log
+
+
+class NewspaperFeed(FeedHandler):
+    def __init__(self, url: str, feed_type: int, use_cache: bool = True):
+        super()
+        print("Created Newspaper Source")
+        self.url = url
+        self.feed_type = feed_type
+        data = newspaper.Source(url)
```

Comment:
I wonder if it would make sense to move this in a separate function named `extract_feed_metadata`. 
there's a parallel also in the RSSFeed handler. 
And the two strategies are different. 
then in the superclass, you should be able to initialize the url, feed_type, use_cache, and then simply call `self.extract_feed_metadata` that will run the corresponding implementatin ?

Reply:
I have added the method extract_feed_metadata to the interface and implemented them each in the sub-classes. So now, when an object is created they call their super constructor and then extract the metadata as you suggested 👍